bouncerscript/task.py

```diff
@@ -59,3 +65,18 @@ def validate_task_schema(context):
     action = get_task_action(context.task, context.config)
     schema_key = "schema_files.{}".format(action)
     client.validate_task_schema(context, schema_key=schema_key)
+
+
+def preflight_check(context):
+    """Make sure we don't do any cross-product/channel alias update"""
+    aliases = context.task["payload"]["aliases_entries"]
+    allowed_regexes = context.config['aliases_regexes']
+
+    validations = []
+    for alias, product_name in aliases.items():
+        if alias not in allowed_regexes.keys():
+            raise TaskVerificationError("Unrecognized alias")
+
+        validations.append(matches(product_name, allowed_regexes[alias]))
+
+    return all(validations)
```

Comment:
Nit: I'm confused by the fact this function returns a boolean:

* the function name doesn't contain regular "boolean" indicators like "is", "does", "can", "should"
* I might be wrong, but I think "check" implies we must stop eveything if it fails.
* the function raises an error for a specific case, already

Therefore, as a matter of consistency, I'd change:

```py
if not all(validations):
    raise TaskVerificationError("The product/alias pairs are corrupt: {}".format(aliases))
```

On the bright side, you don't need to modify the super nice test set you have:

```py
# Keep data as is.
def test_check_product_names_match_aliases(aliases_context, raises, expected):
    context = aliases_context
    context.task["payload"]["aliases_entries"] = entries
   if raises:
      with pytest.raises(TaskValidationError):
          check_product_names_match_aliases(context) == expected
   else:
          check_product_names_match_aliases(context)
```

Reply:
I think this approach is much better, thanks for suggesting it. Amended in follow-up fix.